server_app/user/service.js

```diff
@@ -129,6 +129,54 @@ const resetPassword = async (mail, code, newPassword) => {
   });
 }
 
+const updatePassword = async (username, password, newPassword) => {
+  return new Promise(async (resolve, reject) => {
+    try {
+      const userDocument = await UserRepository.findOne({
+        "login": username,
+        "password": encriptor.Crypto(password, username),
+      });
+
+      if (userDocument == null){
+        return resolve(false);
+      }
+
+      userDocument.password = encriptor.Crypto(newPassword, username);
+      userDocument.save();
+      return resolve(true);
+    } catch (error) {
+      console.error(error);
+      reject(error);
+    }
+  });
+}
+
+const updateUserData = async (userId, newName, newMail ) => {
+  return new Promise(async (resolve, reject) => {
+    try {
+      const userDocument = await UserRepository.findById(userId);
+
+      if (userDocument == null){
+        return resolve(false);
+      }
+
+			if (newMail != ""){
+				userDocument.login = newMail;
+			}
+
+			if (newName != ""){
+				userDocument.name = newName;
+			}
+
+      userDocument.save();
```

Comment:
Here we should validate if there is another user using this mail before we make the change.